pre-commit.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+# In what follows
+#   '! foo | read'
+# means fail if foo has any output, because
+#   cat /dev/null | read
+# exits with $? == 1, and the
+# leading `!` negates that to success.
+
+! gofmt -s -d -l . 2>&1 | read
+! goimports -l . | read
+! go vet -all ./... 2>&1 | read
+! golint ./... | read
+
+go test k8s.io/kubectl/cmd/kinflate/...
```

Comment:
I think it not necessary to have this line, I believe your travis config will auto run all go tests.

Reply:
I wanted to have a pre-commit script i could run manually.
My read of https://docs.travis-ci.com/user/languages/go/#Default-Build-Script
suggests that specifying "script" overrides the default behavior.